api/user.go

```diff
@@ -245,6 +245,11 @@ func CreateUser(user *model.User) (*model.User, *model.AppError) {
 	user.MakeNonNil()
 	user.Locale = *utils.Cfg.LocalizationSettings.DefaultClientLocale
 
+	if err := utils.IsPasswordValid(user.Password); user.AuthService == "" && err != nil {
+		err.StatusCode = http.StatusNotAcceptable
```

Comment:
Not sure we use this status code anywhere else?  I think 500 or invalid arg is good.